sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+constexpr static const char* k_HeaderContentMD5 = "x-ms-content-md5";
+constexpr static const char* k_HeaderUmask = "x-ms-umask";
+constexpr static const char* k_HeaderPermissions = "x-ms-permissions";
+constexpr static const char* k_HeaderRenameSource = "x-ms-rename-source";
+constexpr static const char* k_HeaderOwner = "x-ms-owner";
+constexpr static const char* k_HeaderGroup = "x-ms-group";
+constexpr static const char* k_HeaderAcl = "x-ms-acl";
+constexpr static const char* k_HeaderContentLength = "Content-Length";
+constexpr static const char* k_HeaderDate = "Date";
+constexpr static const char* k_HeaderXMsRequestId = "x-ms-request-id";
+constexpr static const char* k_HeaderXMsVersion = "x-ms-version";
+constexpr static const char* k_HeaderXMsContinuation = "x-ms-continuation";
+constexpr static const char* k_HeaderXMsErrorCode = "x-ms-error-code";
+constexpr static const char* k_HeaderETag = "ETag";
+constexpr static const char* k_HeaderLastModified = "Last-Modified";
+constexpr static const char* k_HeaderXMsNamespaceEnabled = "x-ms-namespace-enabled";
+constexpr static const char* k_HeaderXMsProperties = "x-ms-properties";
+constexpr static const char* k_HeaderAcceptRanges = "Accept-Ranges";
+constexpr static const char* k_HeaderContentRange = "Content-Range";
+constexpr static const char* k_HeaderPathLeaseAction = "x-ms-lease-action";
+constexpr static const char* k_HeaderXMsLeaseDuration = "x-ms-lease-duration";
+constexpr static const char* k_HeaderXMsLeaseBreakPeriod = "x-ms-lease-break-period";
+constexpr static const char* k_HeaderXMsLeaseId = "x-ms-lease-id";
+constexpr static const char* k_HeaderXMsLeaseTime = "x-ms-lease-time";
+constexpr static const char* k_HeaderRange = "Range";
+constexpr static const char* k_HeaderXMsRangeGetContentMd5 = "x-ms-range-get-content-md5";
+constexpr static const char* k_HeaderXMsResourceType = "x-ms-resource-type";
+constexpr static const char* k_HeaderXMsLeaseState = "x-ms-lease-state";
+constexpr static const char* k_HeaderXMsLeaseStatus = "x-ms-lease-status";
+constexpr static const char* k_HeaderXMsContentMd5 = "x-ms-content-md5";
+constexpr static const char* k_HeaderXMsOwner = "x-ms-owner";
+constexpr static const char* k_HeaderXMsGroup = "x-ms-group";
+constexpr static const char* k_HeaderXMsPermissions = "x-ms-permissions";
+constexpr static const char* k_HeaderXMsAcl = "x-ms-acl";
+constexpr static const char* k_HeaderXMsClientRequestId = "x-ms-client-request-id";
+
+// Mode "set" sets POSIX access control rights on files and directories, "modify" modifies one or more POSIX access control rights  that pre-exist on files and directories, "remove" removes one or more POSIX access control rights  that were present earlier on files and directories
+enum class PathSetAccessControlRecursiveMode
+{
+  Set,
+  Modify,
+  Remove,
+  Unknown
+};
+
+static std::string PathSetAccessControlRecursiveModeToString(const PathSetAccessControlRecursiveMode& pathSetAccessControlRecursiveMode)
```

Comment:
non-member static function in header file may not be a good idea.

ditto for other functions

Reply:
I'll change it. But can you elaborate on this?

Reply:
some google result https://stackoverflow.com/questions/42056160/static-functions-declared-in-c-header-files

Reply:
If I'm not reading this wrong I think it's to avoid linkage problem. I'll change them.

Reply:
no, usually it won't cause linkage problem. static functions in .cpp only have internal linkage. By putting static function in a header file, we are actually putting the same function in each .cpp that includes this header file, so every .cpp have the same function definition. But since there's no external linkage symbol for these static functions, there won't be link error. So it usually works, but not in an uncommon way.

Reply:
We may use `inline` instead.